include/exact/ExactMapper.hpp

```diff
@@ -18,6 +19,10 @@
 #include <unordered_set>
 #include <z3++.h>
 
+#ifdef Z3_FOUND
+    #include "LogicBlock/Z3Logic.hpp"
+#endif
+
 using namespace z3;
 using matrix      = std::vector<expr_vector>;
```

Comment:
if z3 is not available, this does not make sense. so this should also be part of an `#ifdef Z3_FOUND` block. if it is needed at all anymore.
in order to comply with the changes above, this `using` directive should also be moved to the corresponding source file.